### PR TITLE
Update layout.html file in sphinx to add top menubar to astropy docs

### PR DESCRIPTION
**Pull request by kakirastern**
Addresses Issue astropy/astropy#8320. Am in the process of implementing changes to add top menubar to astropy docs. Might need help in debugging.

### CODE REVIEW

**Comment by pllim** [comment]
cc @astrofrog

**Comment by kakirastern** [comment]
Thanks, @pllim! Appreciate it

**Comment by kakirastern** [comment]
Hi! I just found out some issues with my PR and need more time to debug. But any suggestions or ideas would be appreciated.

**Comment by astrofrog** [comment]
Looking at the changes, I think most of these are specific to the core package, so I think we should customize the layout template there, not in this theme package (which is used by many packages)

**Comment by bsipocz** [comment]
@astrofrog - good point, it's my bad to forget about the packages that use the template but are not coordinated packages (for those this could be useful, too I think).

**Comment by kakirastern** [comment]
Yup @astrofrog, no problems! It is my first time so I did not check carefully enough before opening this PR, will effect the changes in my astropy/astropy repo then. Cheers!

**Comment by kakirastern** [comment]
Reopened this PR to follow up as suggested in https://github.com/astropy/astropy/pull/8440#pullrequestreview-209927291.

**Comment by kakirastern** [comment]
Hi @astrofrog, I have added an `astropy-custom-menu` as a `html_theme` option to the astropy-sphinx-theme repo in my PR. So now, at least in principal, if I set `html_theme = 'astropy-custom-menu'` instead of the default `html_theme = 'bootstrap-astropy'` option, the desired changes in the menubar should be effected. Please advise if this is what you had in mind as well.

**Comment by kakirastern** [comment]
And give me a brief moment to undo the previous commits regarding the `bootstrap-astropy` theme...

**Comment by kakirastern** [comment]
Think I have cleaned up every unnecessary change made in this PR prior to the last commit. Please review and confirm. Thanks!

**Comment by kakirastern** [comment]
And just to make sure, if I have changed the `stylesheet` in the `theme.conf` file, then I no longer need to add the new theme CSS style sheet to the `layout.html` file? Is my understanding of this correct?

**Comment by kakirastern** [comment]
I think I have managed to add the proposed menubar by configuring the theme at the astropy/astropy repo without having to resort to change the theme here. Therefore, I am closing this PR.

**Comment by kakirastern** [comment]
Hi, I have reopened this PR, and have added in items in the top menubar that would only be on if the sphinx variable astropy_project_menubar is set to True, so that people can opt out say by setting it to False. The PR is ready for a review now, as I was able to use a version of it on an affiliated repo to check locally, and obtained the desired effects. For a screenshot please see the following:

Screenshot 2019-03-15 22 43 02

**Review by astrofrog — CHANGES_REQUESTED** [review]
Thanks! This is on the right track, but at the moment the ``custom.css`` file duplicates a lot of what's in ``bootstrap-astropy.css`` - do we really need to have two separate CSS files? Either we should just have one file, or if we really need to override some CSS in the case where there is a toolbar, we can just override the styles from ``bootstrap-astropy.css``. Ideally we'd just have one file.

**Comment by kakirastern** [comment]
You are welcome @astrofrog! Yes, the `custom.css` is supposed to replace `bootstrap-astropy.css`, that's why the significant overlapping between the two files. But I initiated wanted to check first before making the replacement final.

**Comment by kakirastern** [comment]
Replacement of `bootstrap-astropy.css` file completed.

File: astropy_sphinx_theme/bootstrap-astropy/layout.html [hunk 1]
```diff
@@ -6,6 +6,9 @@
 {# Add the google webfonts needed for the logo #}
 {% block extrahead %}
 <link href='https://fonts.googleapis.com/css?family=Source+Sans+Pro:200,600' rel='stylesheet' type='text/css'/>
+{% if astropy_project_menubar == true %}
+<link type='text/css' href='_static/custom.css' />
+{% endif %}
```
Comment (astrofrog) [inline astropy_sphinx_theme/bootstrap-astropy/layout.html]: @kakirastern - is this needed anymore?

**Comment by astrofrog** [comment]
In addition to the small comment above, can you also rebase this?

File: astropy_sphinx_theme/bootstrap-astropy/layout.html (hunk 1 above)
Comment (kakirastern) [reply astropy_sphinx_theme/bootstrap-astropy/layout.html]: Yup, it is need, for people to opt in or out of the `astropy_project_menubar` as an option.

Comment (kakirastern) [reply astropy_sphinx_theme/bootstrap-astropy/layout.html]: Sorry, I confused this with something else. It is indeed not needed any more and has been removed.

**Comment by kakirastern** [comment]
Yup, rebased and squashed all commits into one and resolved one conflict in the layout.html file.

**Comment by astrofrog** [comment]
@kakirastern - there was an issue with the rebase, and a residual merge commit, so I've rebased it again and pushed to your branch. I think this should then be good to go.

**Comment by astrofrog** [comment]
@kakirastern - the option couldn't currently be set from the ``conf.py`` file so I've fixed that up and also added a changelog entry and information in the README.

**Review by astrofrog — APPROVED** [review]

**Comment by astrofrog** [comment]
I've also removed the ``astropy_banner_96.png`` file which was incorrectly added to this PR.

**Pull request merged**